src/other/mlpmixer.jl

```diff
@@ -0,0 +1,86 @@
+([reference](https://arxiv.org/pdf/2105.01601)).
+
+# Arguments
+- imsize: the size of the input image
+- inchannels: the number of input channels
+- patch_size: the size of the patches
+- planes: the number of channels fed into the main model
+- depth: the number of blocks in the main model
+- expansion_factor: the number of channels in each block
+- dropout: the dropout rate
+- nclasses: the number of classes in the output
+- token_mix: the function to use for the token mixing layer
+- channel_mix: the function to use for the channel mixing layer
+"""
+function mlpmixer(imsize::NTuple{2} = (256, 256); inchannels = 3, patch_size = 16, planes = 512, 
+                  depth = 12, expansion_factor = 4, dropout = 0., nclasses = 1000, token_mix = 
+                  _conv1d, channel_mix = Dense)
+                    
+  im_height, im_width = imsize
+
+  @assert (im_height % patch_size) == 0 && (im_width % patch_size == 0)
+  "image size must be divisible by patch size"
+  
+  num_patches = (im_height ÷ patch_size) * (im_width ÷ patch_size)
+
+  layers = []
+  push!(layers, Patching(patch_size))
+  push!(layers, Dense((patch_size ^ 2) * inchannels, planes))
+  append!(layers, [Chain(_residualprenorm(planes, mlpblock(num_patches, 
+                                          expansion_factor * num_patches, 
+                                          dropout, token_mix)),
+                         _residualprenorm(planes, mlpblock(planes, 
+                                          expansion_factor * planes, dropout, 
+                                          channel_mix)),) for _ in 1:depth])
+
+  classification_head = Chain(_seconddimmean, Dense(planes, nclasses))
+
+  return Chain(layers..., classification_head)
+end
+
+struct MLPMixer
+  layers
+end
+
+"""
+    MLPMixer(imsize::NTuple{2} = (256, 256); inchannels = 3, patch_size = 16, planes = 512, 
+             depth = 12, expansion_factor = 4, dropout = 0., pretrain = false, nclasses = 1000)
+
+Creates a model with the MLPMixer architecture.
+([reference](https://arxiv.org/pdf/2105.01601)).
+
+# Arguments
+- imsize: the size of the input image
+- inchannels: the number of input channels
+- patch_size: the size of the patches
+- planes: the number of channels fed into the main model
+- depth: the number of blocks in the main model
+- expansion_factor: the number of channels in each block
+- dropout: the dropout rate
+- pretrain: whether to load the pre-trained weights for ImageNet
+- nclasses: the number of classes in the output
+"""
+function MLPMixer(imsize::NTuple{2} = (256, 256); inchannels = 3, patch_size = 16, planes = 512, 
+                  depth = 12, expansion_factor = 4, dropout = 0., pretrain = false, nclasses = 1000)
+                    
+  layers = mlpmixer(imsize; inchannels, patch_size, planes, depth, expansion_factor, dropout, 
+                    nclasses)
+  pretrain && loadpretrain!(layers, string("MLPMixer"))
+  MLPMixer(layers)
+end
```

Comment:
Shouldn't we have a separate method for `MLPMixer(; pretrain = false)` that doesn't have all the toggles available. I guess generally the whole repo needs some refactoring to check compatibility for pretrained models.

Reply:
Given that this is a pretty niche model anyways, I'm removing the `pretrain` option entirely for now - when there's a consensus on what to do for the other models we can always come back and fix this one